components/raftstore/src/store/worker/read.rs

```diff
@@ -264,7 +307,7 @@ where
     kv_engine: E,
     metrics: ReadMetrics,
     // region id -> ReadDelegate
-    delegates: HashMap<u64, Option<ReadDelegate>>,
+    delegates: HashMap<u64, Arc<ReadDelegate>>,
```

Comment:
Can it cause memory leak? For example, after many round of splitting and transferring replicas, some peer may be deleted while local reader still keeps a copy of delegate.

Reply:
Yes, but it is not caused by changing `Option` to `Arc`. A peer will call `mark_invalid` when it is destroyed, and when the local reader handles requests, it will check its local `ReadDelegate`'s `invalid` flag and remove the `ReadDelegate` if the flag is true, but if there is no incoming request the local `ReadDelegate` will not be removed.

Reply:
Done, PTAL https://github.com/tikv/tikv/issues/9469